src/Avalonia.Base/CollectionPolyfills.cs

```diff
@@ -0,0 +1,33 @@
+﻿#if !NET6_0_OR_GREATER
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Avalonia
+{
+    internal static class CollectionPolyfills
```

Comment:
Can be moved to Avalonia.Base\Compatibility folder, as we have some similar helpers there already